src/hotspot/share/gc/shenandoah/shenandoahMmuTracker.hpp

```diff
@@ -50,16 +50,32 @@ class ShenandoahMmuTask;
  * MMU.
  */
 class ShenandoahMmuTracker {
+private:
+  // For reporting utilization during most recent GC cycle
+  double _most_recent_timestamp;
+  double _most_recent_gc_time;
+  double _most_recent_gcu;
+  double _most_recent_mutator_time;
+  double _most_recent_mu;
+
+  // For periodic MU/GCU reports
+  double _most_recent_periodic_time_stamp;
+  double _most_recent_periodic_gc_time;
+  double _most_recent_periodic_mutator_time;
+
+  uint _most_recent_gcid;
+  uint _active_processors;
 
-  double _generational_reference_time_s;
-  double _process_reference_time_s;
-  double _collector_reference_time_s;
+  bool _most_recent_is_full;
+  bool _doing_mixed_evacuations;
 
   ShenandoahMmuTask* _mmu_periodic_task;
   TruncatedSeq _mmu_average;
 
-  static double gc_thread_time_seconds();
   static double process_time_seconds();
+  static void fetch_cpu_times(double &gc_time, double &mutator_time);
+
+  void help_record_concurrent(ShenandoahGeneration* generation, uint gcid, const char* msg);
```

Comment:
Do you want to call this `record_concurrent_helper()` or `..._work()` in line with existing helper method convention?

Reply:
Thanks.  Will change this name on expand-old-on-demand branch.